hack/generator/pkg/codegen/storage/property_converter.go

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package storage
+
+import (
+	"fmt"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+)
+
+// PropertyConverter is used to convert the properties of object types as required for storage variants
+type PropertyConverter struct {
```

Comment:
minor, naming: Is this a `PropertyConverter` or a `PropertyConversionBuilder` or `Applier` or something?

It's not actually _doing_ conversions? It's just putting conversions onto types?

Reply:
It changes the types of properties passed to `ConvertProperty()` - it does the conversion.